lib/datadog/appsec/utils/http/media_range.rb

```diff
@@ -0,0 +1,201 @@
+# typed: ignore
+
+require_relative 'media_type'
+
+module Datadog
+  module AppSec
+    module Utils
+      module HTTP
+        # Implementation of media range for content negotiation
+        class MediaRange
+          class ParseError < ::StandardError
+          end
```

Comment:
Wow this is a big addition just to support `Response#format`. I hesistate to ask but... is there any way we could we keep some of the simplicity of the old code?

Reply:
Yeah, I tried. The thing is, the old code was not correct WRT what real-life UAs might send.

So ultimately my choice was either to pile up some more hacks that may still be incomplete in face of whatever UAs do, or implement the real thing.

This is also on a feature that is implemented but currently gated by an undocumented, experimental setting, so the actual risk to the change here is low.